hschain-PoW/HSChain/PoW/Node.hs

```diff
@@ -77,12 +77,13 @@ data Cfg = Cfg
 -- we are mining and genesis block.
 runNode :: forall b s . (BlockData b, Mineable b, Show (b Identity), Serialise (b Proxy), Serialise (b Identity))
         => [String] -> Bool -> Block b
-        -> (Block b -> s -> Maybe s)
-        -> (BH b -> s -> ((Header b, [Tx b]), s))
-        -> (Header b -> [Tx b] -> IO (Maybe (Block b)))
-        -> s
+        -> (Block b -> s -> Maybe s) -- ^ User state step function - update user state with block. May fail.
```

Comment:
Since we decided to not to store mempool in the  StateView changes to runNode should be reverted. Besides #662 reworks runNode quite a bit